pkg/master/master.go

```diff
@@ -167,6 +170,9 @@ type thirdPartyEntry struct {
 	group   unversioned.APIGroup
 }
 
+type RESTOptionsGetter func(resource unversioned.GroupResource) generic.RESTOptions
+type RESTStorageProvider func(apiResourceConfigSource genericapiserver.APIResourceConfigSource, restOptionsGetter RESTOptionsGetter) (genericapiserver.APIGroupInfo, bool)
```

Comment:
> Would this be clearer as an interface? (thinking out loud, not asking for a change)

Yes, having written these, I think I agree, but I'd like to do it as a followup along with splitting apart the individual RESTStorage per-group.